components/blocks/Recipe.tsx

```diff
@@ -0,0 +1,320 @@
+          style={{
+            height:
+              typeof window !== 'undefined' && window.innerWidth >= 1024
+                ? `${LHSheight}px`
+                : `${smAndMbHeight}`,
+          }}
+        >
+          <div className={`${isBottomOfInstructions ? 'hidden' : ''}`}>
+            <div
+              className={`absolute inset-0 bg-gradient-to-b from-transparent via-transparent to-black opacity-60 lg:rounded-bl-xl pointer-events-none `}
+            ></div>
+            <FaChevronCircleDown
+              onClick={handleDownArrowClick}
+              className={`absolute bottom-4 left-1/2 transform -translate-x-1/2 w-7 h-7 text-xl text-white cursor-pointer shadow-md`}
+            />
+          </div>
+
+          <div
+            className="overflow-auto rounded-tl-xl rounded-bl-xl rounded-tr-xl lg:rounded-tr-none"
+            onScroll={checkIfBottom}
+          >
+            {instruction?.map((inst, idx) => (
+              <div
+                key={idx}
+                ref={(el) => (instructionRefs.current[idx] = el)}
+                className={`instruction-item cursor-pointer p-4 border-gray-700 border-y bg-gray-800 text-white 
+                ${clickedInstruction === idx ? 'bg-slate-600' : ''} `}
+                onClick={() =>
+                  handleInstructionClick(
+                    idx,
+                    inst.codeLineStart,
+                    inst.codeLineEnd
+                  )
+                }
+              >
+                <h5 className="font-tuner">{`${idx + 1}. ${
+                  inst.header || 'Default Header'
+                }`}</h5>
+                <div
+                  className={`overflow-auto transition-all duration-500 ease-in-out ${
+                    clickedInstruction === idx
+                      ? 'max-h-full opacity-100'
+                      : 'max-h-0 opacity-0'
+                  }`}
+                >
+                  <span className="mt-2">
+                    {inst.itemDescription || 'Default Item Description'}
+                  </span>
+                </div>
+              </div>
+            )) || <p>No instructions available.</p>}
+          </div>
+        </div>
+
+        <div
+          ref={codeblockRef}
+          className="codeblock bg-gray-800 lg:w-2/3 max-h-50vh overflow-auto lg:rounded-tr-xl rounded-bl-xl lg:rounded-bl-none rounded-br-xl "
+        >
+          {codeblock ? (
+            <div>
+              <TinaMarkdown
+                key={highlightLines}
+                content={codeblock}
+                components={{
+                  code_block: (props) => (
+                    <CodeBlockWithHighlightLines
+                      {...props}
+                      highlightLines={highlightLines}
+                    />
+                  ),
+                }}
+              />
+            </div>
+          ) : (
+            <p>No code block available.</p>
+          )}
```

Comment:
cc @JackDevAU 
I dont like that we need to have a `TinaMarkdown` rich text field here to get the code block.

My suggested solution
1. New custom Tina field - just a string but with a `ui` override with a nice code editor
2. use that here instead of `TinaMarkdown`

<img width="1782" alt="image" src="https://github.com/user-attachments/assets/595175cd-91ac-447d-98ab-766ce2adff14">

Reply:
Done, new monaco editor has been added 

![Screenshot 2024-11-08 at 10 19 13 am](https://github.com/user-attachments/assets/8dd7396f-0fbb-4eac-a295-e66013ed8897)
**Figure: New Monaco Editor**